jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java

```diff
@@ -389,7 +398,12 @@ public void succeeded()
         if (process)
         {
             onSuccess();
-            processing();
+            boolean processing;
+            try (AutoLock ignored = _lock.lock())
+            {
+                processing = _state == State.PROCESSING;
+            }
+            processing(processing);
```

Comment:
I would also like to see a test covering this case.

Where the `succeeded()` is called externally to the iteration of the ICB.
```java
icb.iterate();
assertEquals(1, count.get());
icb.succeeded();
assertTrue(icb.isAborted());
```